File: migoo_crm/migoo_crm/report/lead_summary/lead_summary.js
// Copyright (c) 2023, Palak Padalia and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Lead Summary"] = {
	"filters": [
		{
			"fieldname": "creation",
			"label": __("Period"),
			"fieldtype": "Select",
			"options": "Daily\nTill Date\nDo Not Contact\nTotal",
		},
		{
			"fieldname": "from_date",
			"label": __("From Date"),
			"fieldtype": "Date",
			'depends_on': 'eval:doc.creation === "Till Date"',

		},
		{
			"fieldname": "to_date",
			"label": __("To Date"),
			"fieldtype": "Date",
			'depends_on': 'eval:doc.creation === "Till Date"',
		}
	]
};

File: migoo_crm/migoo_crm/report/lead_summary/lead_summary.py
# Copyright (c) 2023, Palak Padalia and contributors
# For license information, please see license.txt

import frappe
from frappe import _
import datetime


@frappe.whitelist()
def execute(filters=None):
    columns = [{
        "label": _("Name"),
        "fieldname": "lead_name",
        "width": 130,
    }]
    data = []

    # Check the value of the filter
    period = filters.get("creation")
    from_date = filters.get("from_date")
    to_date = filters.get("to_date")

    query = '''
    select lead_name
    from `tabLead`
    '''
    if period == "Daily":
        query += " WHERE DATE(creation) = CURDATE()"
    elif period == "Till Date":
        if from_date and to_date:
            query += f" WHERE DATE(creation) >= '{from_date}' AND DATE(creation) <= '{to_date}'"
        else:
            last_date = datetime.date.today() - datetime.timedelta(days=1)  # Get the last date
            query += f" WHERE DATE(creation) <= '{last_date}'"

    elif period == "Till Date":
        pass

    elif period == "Do Not Contact":
        query += " WHERE status='Do Not Contact'"

    data = frappe.db.sql(query, as_dict=1)

    return columns, data
